src/libraries/System.Security.Cryptography.Encoding/src/System/Security/Cryptography/PemEncoding.cs

```diff
@@ -0,0 +1,518 @@
+                // label, so move from there.
+                if (!IsValidLabel(label))
+                {
+                    goto NextAfterLabel;
+                }
+
+                int contentStartIndex = preebEndIndex + Ending.Length;
+                int postebLength = PostEBPrefix.Length + label.Length + Ending.Length;
+
+                Span<char> postebBuffer = postebLength > PostebStackBufferSize
+                    ? new char[postebLength]
+                    : postebStackBuffer;
+                ReadOnlySpan<char> posteb = WritePostEB(label, postebBuffer);
+                int postebStartIndex = pemData.IndexOfByOffset(posteb, contentStartIndex);
+
+                if (postebStartIndex < 0)
+                {
+                    goto NextAfterLabel;
+                }
+
+                int pemEndIndex = postebStartIndex + postebLength;
+
+                // The PostEB must either end at the end of the string, or
+                // have at least one white space character after it.
+                if (pemEndIndex < pemData.Length - 1 &&
+                    !char.IsWhiteSpace(pemData[pemEndIndex]))
+                {
+                    goto NextAfterLabel;
+                }
+
+                Range contentRange = contentStartIndex..postebStartIndex;
+
+                if (!TryCountBase64(pemData[contentRange],
+                                   out int base64start,
+                                   out int base64end,
+                                   out int decodedSize))
+                {
+                    goto NextAfterLabel;
+                }
+
+                Range pemRange = preebIndex..pemEndIndex;
+                Range base64range = (contentStartIndex + base64start)..
+                                    (contentStartIndex + base64end);
+                fields = new PemFields(labelRange, base64range, pemRange, decodedSize);
+                return true;
+
+                NextAfterLabel:
+                if (preebEndIndex <= 0)
+                {
+                    // We somehow ended up in a situation where we will advance
+                    // 0 or -1 characters, which means we'll probably end up here again,
+                    // advancing 0 or -1 characters, in a loop. To avoid getting stuck,
+                    // detect this situation and return.
+                    fields = default;
+                    return false;
+                }
+                areaOffset += preebEndIndex;
+            }
+
+            fields = default;
+            return false;
+
+            static ReadOnlySpan<char> WritePostEB(ReadOnlySpan<char> label, Span<char> destination)
+            {
+                int size = PostEBPrefix.Length + label.Length + Ending.Length;
+                Debug.Assert(destination.Length >= size);
+                PostEBPrefix.AsSpan().CopyTo(destination);
+                label.CopyTo(destination[PostEBPrefix.Length..]);
```

Comment:
I think that the left-defined-right-open calls look much nicer as explicitly saying `destination.Slice(PostEBPrefix.Length)`.

Maybe I'll feel different in a few years :)

Reply:
> Maybe I'll feel different in a few years :)

I kind of agree but felt weird mixing range-style slicing and literally calling `Slice`.

> I think that the left-defined-right-open calls look much nicer as explicitly saying 

Is this a stream of consciousness thought, or a "please change `[X..]` to `Slice(x)`??

Reply:
> Is this a stream of consciousness thought, or a "please change [X..] to Slice(x)??

An "I'd prefer" :smile:.

We don't have a style rule or strong convention; but some of us have a general leeriness of the syntax because `str[a..b]` is Substring, and `arr[a..b]` is effectively `Slice().ToArray()` -- the decision for range indexers is they return a collection of the same type, not a Span/Memory-equivalent projection.  For Span/ROSpan that means it's "zero"-cost slicing, but as syntax it's a bit of a burden on reviewers.

My personal filter is that I can accept the reviewer burden on "wait, are we a span type already?" for `[start..end]` (to avoid changing it to start/count); but for `[start..]` (`Slice(start)`) and `[..end]` (`Slice(0, end)`) the call to Slice reduces my cognitive load without adding increased load at changing parameter semantics.  (I'm also happy with changing `[start..end]` to `Slice(start, end - start)`, but it's more contextual)

Reply:
Seems reasonable. I removed the open-ended Range syntax in favor of explicit Slice. It's much to my preference to keep the [x..y] syntax for the reason you described.